examples/add_points_on_nD_shapes.py

```diff
@@ -0,0 +1,66 @@
+"""Add points on nD shapes in 3D using a mouse callback"""
+import napari
+import numpy as np
+
+# Create rectangles in 4D
+shapes_data = np.array(
+    [
+        [
+            [0, 50, 75, 75],
+            [0, 50, 125, 75],
+            [0, 100, 125, 125],
+            [0, 100, 75, 125]
+        ],
+        [
+            [0, 10, 75, 75],
+            [0, 10, 125, 75],
+            [0, 40, 125, 125],
+            [0, 40, 75, 125]
+        ],
+        [
+            [1, 100, 75, 75],
+            [1, 100, 125, 75],
+            [1, 50, 125, 125],
+            [1, 50, 75, 125]
+        ]
+    ]
+)
+
+# add an empty 4d points layer
+viewer = napari.view_points(ndim=4, size=3)
+points_layer = viewer.layers[0]
+
+# add the shapes layer to the viewer
+properties = {'index': [0, 1, 2]}
+shapes_layer = viewer.add_shapes(
+    shapes_data,
+    face_color=['magenta', 'green', 'blue'],
+    edge_color='white',
+    blending='additive',
+    properties=properties,
+    text='index'
+)
+
+
+@shapes_layer.mouse_drag_callbacks.append
+def on_click(layer, event):
+
+    start_point, end_point = layer.get_ray_intersections(
+        event.position,
+        event.view_direction,
+        event.dims_displayed
+    )
+    if (start_point is not None) and (end_point is not None):
```

Comment:
After the discussion in the euro dev meeting yesterday, I will refactor the convenience method to take the properties of the mouse event as kwargs instead of the event object itself.